File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BrandingColorsModel.java
package kore.botssdk.models;

import kore.botssdk.utils.StringUtils;

public class BrandingColorsModel {
    String primary;
    String secondary;
    String primary_text;
    String secondary_text;
    Boolean useColorPaletteOnly;

    public BrandingColorsModel updateWith(BrandingColorsModel configModel) {
        primary = !StringUtils.isNullOrEmpty(configModel.primary) ? configModel.primary : primary;
        secondary = !StringUtils.isNullOrEmpty(configModel.secondary) ? configModel.secondary : secondary;
        primary_text = !StringUtils.isNullOrEmpty(configModel.primary_text) ? configModel.primary_text : primary_text;
        secondary_text = !StringUtils.isNullOrEmpty(configModel.secondary_text) ? configModel.secondary_text : secondary_text;
        useColorPaletteOnly = configModel.useColorPaletteOnly != null ? configModel.useColorPaletteOnly : useColorPaletteOnly;
        return this;
    }

    public String getPrimary() {
        return primary;
    }

    public String getPrimary_text() {
        return primary_text;
    }

    public String getSecondary() {
        return secondary;
    }

    public String getSecondary_text() {
        return secondary_text;
    }

    public boolean isUseColorPaletteOnly() {
        return useColorPaletteOnly != null ? useColorPaletteOnly : false;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BrandingColorsModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class BrandingColorsModel(
    @SerializedName("primary")
    val primary: String? = null,
    @SerializedName("secondary")
    val secondary: String? = null,
    @SerializedName("primary_text")
    val primaryText: String? = null,
    @SerializedName("secondary_text")
    val secondaryText: String? = null,
    @SerializedName("useColorPaletteOnly")
    val useColorPaletteOnly: Boolean? = null
)

fun BrandingColorsModel.updateWith(configModel: BrandingColorsModel): BrandingColorsModel {
    return this.copy(
        primary = configModel.primary?.ifEmpty { this.primary } ?: primary,
        secondary = configModel.secondary?.ifEmpty { this.secondary } ?: secondary,
        primaryText = configModel.primaryText?.ifEmpty { this.primaryText } ?: primaryText,
        secondaryText = configModel.secondaryText?.ifEmpty { this.secondaryText } ?: secondaryText,
        useColorPaletteOnly = configModel.useColorPaletteOnly ?: this.useColorPaletteOnly,
    )
}